src/containers/audio-rewards-page/components/modals/ChallengeRewards.tsx

```diff
@@ -209,9 +220,45 @@ const ChallengeRewardsBody = ({ dismissModal }: BodyProps) => {
     </div>
   )
 
+  const { toast } = useContext(ToastContext)
+  const claimStatus = useSelector(getClaimStatus)
+  const [hideClaimButton, setHideClaimButton] = useState(false)
+  const [displayClaimError, setDisplayClaimError] = useState(false)
+  const [icon, setIcon] = useState(<IconCheck />)
```

Comment:
probably shouldn't put the component itself in state. i think that may have some odd render consequences
https://stackoverflow.com/a/53976730

an enum or similar would be best and conditionally passing the icon down based on that value